web/src/main/java/se/inera/intyg/webcert/web/event/CertificateEventLoaderServiceImpl.java

```diff
@@ -0,0 +1,132 @@
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import se.inera.intyg.webcert.persistence.event.model.CertificateEventProcessed;
+import se.inera.intyg.webcert.persistence.event.repository.CertificateEventProcessedRepository;
+import se.inera.intyg.webcert.persistence.utkast.repository.UtkastRepository;
+import se.inera.intyg.webcert.web.service.monitoring.MonitoringLogService;
+
+@Service
+@EnableScheduling
+public class CertificateEventLoaderServiceImpl implements CertificateEventLoaderService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(CertificateEventLoaderServiceImpl.class);
+    private static final String JOB_NAME = "CertificateEventLoader.run";
+    private static final int NR_OF_BATCHES = 32;
+
+    @Autowired
+    MonitoringLogService monitoringLogService;
+
+    @Autowired
+    UtkastRepository utkastRepository;
+
+    @Autowired
+    CertificateEventProcessedRepository processedRepository;
+
+    @Autowired
+    @Qualifier("jmsCertificateEventLoaderTemplate")
+    private JmsTemplate jmsTemplate;
+
+    @Value("${certificateeventloader.batchsize:10000}")
+    private Integer batchSize;
+
+    @Value("${certificateevent.loader.queueName}")
+    private String internalCertificateEventsLoaderQueue;
+
+
+    @Override
+    @Scheduled(cron = "${certificateeventloader.cron:-}")
+    @SchedulerLock(name = JOB_NAME)
+    public void loadIds() {
+        if (countPendingMessages() == 0) {
+            LOG.info("No IDs on queue: Starting certificate event loader run with batch size: " + batchSize + " and splitting into "
+                + NR_OF_BATCHES + " batches.");
+            var certificateIdList = getIdsForCertificatesWithoutEvents();
+
+            if (certificateIdList.size() > 0) {
+                var size = certificateIdList.size() < NR_OF_BATCHES ? NR_OF_BATCHES : (certificateIdList.size() / NR_OF_BATCHES);
+                chunked(certificateIdList.stream(), size).forEach(this::putIdsOnQueue);
+            }
+        }
+    }
+
+    private static <T> Stream<List<T>> chunked(Stream<T> stream, int chunkSize) {
+        AtomicInteger index = new AtomicInteger(0);
+
+        return stream.collect(Collectors.groupingBy(x -> index.getAndIncrement() / chunkSize))
+            .entrySet().stream()
+            .sorted(Map.Entry.comparingByKey()).map(Map.Entry::getValue);
+    }
+
+    private int countPendingMessages() {
+        Integer totalPendingMessages = this.jmsTemplate
+            .browse(internalCertificateEventsLoaderQueue, (session, browser) -> Collections.list(browser.getEnumeration()).size());
+
+        return totalPendingMessages == null ? 0 : totalPendingMessages;
+    }
+
+    @Transactional
+    public void putIdsOnQueue(List<String> idList) {
+        var success = send(session -> session.createObjectMessage((ArrayList<String>) idList));
+        if (success) {
+            idList.forEach(this::addToProcessed);
```

Comment:
Thinking how this will perform. Of course this is relative small amount of time considering creating actual events for each certificate, but it still could add up. Googled a little bit and found this interesting question with answers with ideas/options: https://stackoverflow.com/questions/44243608/need-to-insert-100000-rows-in-mysql-using-hibernate-in-under-5-seconds